android/src/main/java/im/status/ethereum/keycard/SmartCard.java

```diff
@@ -325,8 +329,10 @@ public WritableMap getKeys(final String pairingBase64, final String pin) throws
         ApplicationInfo info = new ApplicationInfo(cmdSet.select().checkOK().getData());
 
         WritableMap data = Arguments.createMap();
-        data.putString("address", Hex.toHexString(keyPair.toEthereumAddress()));
-        data.putString("public-key", Hex.toHexString(keyPair.getPublicKey()));
+        data.putString("address", Hex.toHexString(masterPair.toEthereumAddress()));
+        data.putString("public-key", Hex.toHexString(masterPair.getPublicKey()));
```

Comment:
I understand that we use the master key address to have a deterministic id, what do we need the public key for?

Reply:
@gravityblast I bet it is not really used anywhere, but it is still a part of some data structures on cljs side. It is not persisted. This pr is just to align on-device and keycard acc, and will file a separate later for status-react and this repo to clean up unused fields.